File: src/fizzbuzz.pas
program FizzBuzz;
uses sysutils;

// FizzBuzzの値を返す関数 --- (*1)
function GetFizzBuzz(i: Integer): String;
  // インライン関数 --- (*2)
  function IsFizz(i: Integer): Boolean;
  begin
    Result := (i Mod 3 = 0);
  end;
  function IsBuzz(i: Integer): Boolean;
  begin
    Result := (i Mod 5 = 0);
  end;
begin
  if IsFizz(i) and IsBuzz(i) then Result := 'FizzBuzz'
  else if IsFizz(i) then Result := 'Fizz'
  else if IsBuzz(i) then Result := 'Buzz'
  else Result := IntToStr(i);
end;

// メイン処理 --- (*3)
var
  i: Integer;
  s: String;
begin
  for i := 1 to 100 do begin
    s := GetFizzBuzz(i);
    Writeln(s);
  end;
end.

File: src/fizzbuzz.f90
PROGRAM fizzbuzz
        IMPLICIT NONE ! 暗黙の型宣言を禁止 --- (*1)
        INTEGER :: i  ! 変数iの宣言 --- (*2)
        ! 1から100まで繰り返す --- (*3)
        DO i = 1, 100
                ! 順次判定していく --- (*4)
                IF (MOD(i, 3) == 0 .AND. MOD(i, 5) == 0) THEN
                        PRINT *, "FizzBuzz"
                ELSE IF (MOD(i, 3) == 0) THEN
                        PRINT *, "Fizz"
                ELSE IF (MOD(i, 5) == 0) THEN
                        PRINT *, "Buzz"
                ELSE
                        PRINT *, i
                END IF
        END DO
END PROGRAM fizzbuzz

